nms/app/packages/magmalte/app/components/feg/FEGContext.js

```diff
@@ -0,0 +1,103 @@
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+import * as React from 'react';
+import FEGNetworkContext from '../context/FEGNetworkContext';
+import LoadingFiller from '@fbcnms/ui/components/LoadingFiller';
+import MagmaV1API from '@fbcnms/magma-api/client/WebClient';
+
+import type {feg_network, network_id, network_type} from '@fbcnms/magma-api';
+
+import {UpdateNetworkState as UpdateFegNetworkState} from '../../state/feg/NetworkState';
+
+import {useEffect, useState} from 'react';
+import {useEnqueueSnackbar} from '@fbcnms/ui/hooks/useSnackbar';
+
+type Props = {
+  networkId: network_id,
+  networkType: network_type,
+  children: React.Node,
+};
+import {FEG} from '@fbcnms/types/network';
+
+/**
+ * Fetches and returns information about the federation network inside
+ * a context provider.
+ * @param {object} props: contains the network id and its type
+ */
+export function FEGNetworkContextProvider(props: Props) {
+  const {networkId} = props;
+  const [fegNetwork, setFegNetwork] = useState<$Shape<feg_network>>({});
+  const [isLoading, setIsLoading] = useState(true);
+  const enqueueSnackbar = useEnqueueSnackbar();
+
+  useEffect(() => {
+    const fetchState = async () => {
+      try {
+        setFegNetwork(await MagmaV1API.getFegByNetworkId({networkId}));
+      } catch (e) {
+        enqueueSnackbar?.('failed fetching network information', {
+          variant: 'error',
+        });
+      }
+      setIsLoading(false);
+    };
+    fetchState();
+  }, [networkId, enqueueSnackbar]);
+
+  if (isLoading) {
+    return <LoadingFiller />;
+  }
+
+  return (
+    <FEGNetworkContext.Provider
+      value={{
+        state: fegNetwork,
+        updateNetworks: props => {
+          let refreshState = true;
+          if (networkId !== props.networkId) {
+            refreshState = false;
+          }
+          return UpdateFegNetworkState({
+            networkId,
+            setFegNetwork,
+            refreshState,
+            ...props,
+          });
+        },
+      }}>
+      {props.children}
+    </FEGNetworkContext.Provider>
+  );
+}
+/**
+ * A context provider for federation networks. It is used in sharing
+ * information like the network information or the gateways information.
+ * @param {object} props contains the network id and its type
+ */
+export function FEGContextProvider(props: Props) {
+  const {networkId, networkType} = props;
+  const fegNetwork = networkType === FEG;
+  if (!fegNetwork) {
+    return props.children;
```

Comment:
Oh, I forgot to remove that once I created a network context provider. Since the network context provider now take care of that, I have removed it.

Reply:
Oh, I forgot to remove that once a network context provider was added. It is not valid anymore since the network context provider takes care of that. I have removed it. Thanks!